src/OpenTelemetry.AutoInstrumentation/Configurations/FileBasedConfiguration/ResourceConfiguration.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Vendors.YamlDotNet.Serialization;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+
+internal class ResourceConfiguration
+{
+    /// <summary>
+    /// Gets or sets the list of resource attributes.
+    /// </summary>
+    [YamlMember(Alias = "attributes")]
+    public List<ResourceAttribute>? Attributes { get; set; }
+
+    /// <summary>
+    /// Gets or sets the attributes list for the resource.
+    /// </summary>
+    [YamlMember(Alias = "attributes_list")]
+    public string? AttributesList { get; set; }
+
+    public List<KeyValuePair<string, object>> ParseAttributes()
+    {
+        var resourceAttributesWithPriority = new Dictionary<string, object>();
+
+        if (Attributes != null)
+        {
+            foreach (var attr in Attributes)
+            {
+                if (!resourceAttributesWithPriority.ContainsKey(attr.Name))
+                {
+                    // TODO parse type and converting the value accordingly.
+                    resourceAttributesWithPriority.Add(attr.Name, attr.Value);
+                }
+            }
+        }
+
+        if (AttributesList != null)
+        {
+            const char attributeListSplitter = ',';
+            const char attributeKeyValueSplitter = '=';
+
+            var rawAttributes = AttributesList.Split(attributeListSplitter);
+            foreach (var rawKeyValuePair in rawAttributes)
+            {
+                var keyValuePair = rawKeyValuePair.Split(attributeKeyValueSplitter);
```

Comment:
Nit: May cause issues with base64 values. Not sure we use any, specially here. Easy fix would be (since keys never use `=`):

```suggestion
                var keyValuePair = rawKeyValuePair.Split([attributeKeyValueSplitter], count: 2);
```

Maybe worth adding `StringSplitOptions.RemoveEmptyEntries`.